File: articles/resilience-thinking-disturbance-adaptation-thresholds/c/viability_model.c
#include <stdio.h>

int main(void) {
    const int steps = 30;
    double viability[steps];
    double disturbance = 0.08;
    double adaptive_response = 0.09;

    viability[0] = 1.0;

    for (int t = 1; t < steps; t++) {
        viability[t] = viability[t - 1] - disturbance + adaptive_response;

        if (viability[t] < 0.0) {
            viability[t] = 0.0;
        }

        if (viability[t] > 1.5) {
            viability[t] = 1.5;
        }
    }

    printf("Viability simulation\n");

    for (int t = 0; t < steps; t++) {
        printf("time=%d, viability=%.3f\n", t, viability[t]);
    }

    return 0;
}

File: articles/resilience-thinking-disturbance-adaptation-thresholds/fortran/viability_model.f90
program viability_model
  implicit none

  integer, parameter :: steps = 30
  real :: viability(steps)
  real :: disturbance, adaptive_response
  integer :: t

  viability(1) = 1.0
  disturbance = 0.08
  adaptive_response = 0.09

  do t = 2, steps
     viability(t) = viability(t - 1) - disturbance + adaptive_response
     viability(t) = min(1.5, max(0.0, viability(t)))
  end do

  print *, "Viability simulation"

  do t = 1, steps
     print *, t, viability(t)
  end do

end program viability_model
